src/components/Auction/Auction.jsx

```diff
@@ -83,6 +102,35 @@ class Auction extends Component {
     );
   }
 
+  onTabClick = async index => {
+    const { activeTabIndex } = this.state;
+
+    if (index === activeTabIndex) {
+      return null;
+    }
+    if (index === 0) {
+      this.setState({
+        data: this.props.circulatingNecData,
+      });
+    } else if (index === 1) {
+      this.setState({
+        data: this.props.burnedNecData,
+      });
+    } else if (index === 2) {
+      this.setState({
+        data: this.props.deversifiNecEthData,
+      });
+    } else if (index === 3) {
+      this.setState({
+        data: this.props.nextAuctionEthData,
+      });
+    }
```

Comment:
I guess switch statement would be slightly neater

Reply:
I am not happy with this either and I am not a fan of switch case so this will be replaced by something else in the next commit.